### DISCUSSION TITLE
Hide git status in certain repos

### DISCUSSION THREAD

**Discussion by PatrickF1** [starship/starship#1961] [Q&A] [answered]
Hi, my apologies if this has already been asked before--I have thoroughly searched starship issues and discussions. When I am in a large repo, the git status module runs very slowly so I want it disabled. But in normal repos, I want git status. How do I configure starship to only disable git status in certain repos? Thank you.

**Comment by mickimnet** [comment, marked as answer]
That feature is currently not available. Fore more information, please take a look at issue #1781.

Besides, you could edit your prompt configuration to check for the current directory and than, either working with `starship config git_branch.disabled true` (and the other git modules) or using a different `starship.toml` configuration file.

**Reply by PatrickF1** [reply]
Hi @mickimnet, thanks so much for your reply. I hope that feature will come soon! In the meantime, let me make sure I understand your suggestion. Are you suggesting I use a custom command module to execute `starship config git_branch.disabled true` when in a certain directory?

**Reply by matchai** [reply]
I think a tool better suited for such changes would be [direnv](https://direnv.net/), which folks use to change their shell configuration per-directory.
Keep in mind that running `starship config` commands would affect all shells you currently have open.

A preferable solution may be to have direnv set the `STARSHIP_CONFIG` environment variable to the directory of an alternative `starship.toml` config file, as suggested by Mick.

**Reply by PatrickF1** [reply]
Thanks for jumping in @matchi! And also for being the owner of starship! I followed you from spacefish so thank you for that as well.

Anyway, in light of what you said, am I right to expect #1781 to not be implemented anytime soon?

**Reply by mickimnet** [reply]
It is in the [Milestones](https://github.com/starship/starship/milestones/v1.0.0) for the v1 release, so it may take a bit.
